DatadogInternal/Sources/Telemetry/Telemetry.swift

```diff
@@ -75,20 +96,28 @@ public protocol Telemetry {
 }
 
 public extension Telemetry {
-    /// Starts a method call.
+    /// Starts timing a method call using the "Method Called" metric.
     ///
     /// - Parameters:
-    ///   - operationName: Platform agnostic name of the operation.
-    ///   - callerClass: The name of the class that calls the method.
-    ///   - samplingRate: The sampling rate of the method call. Value between `0.0` and `100.0`, where `0.0` means NO event will be processed and `100.0` means ALL events will be processed. Note that this value is multiplicated by telemetry sampling (by default 20%) and metric events sampling (hardcoded to 15%). Making it effectively 3% sampling rate for sending events, when this value is set to `100`.
+    ///   - operationName: A platform-agnostic name for the operation.
+    ///   - callerClass: The name of the class that invokes the method.
+    ///   - sampleRate: The sample rate for **head-based** sampling of the method call metric. Must be a value between `0` (reject all) and `100` (keep all).
+    ///
+    /// Note: The head sample rate is compounded with the tail sample rate, which is configured in `stopMethodCalled()`. Both are applied
+    /// in addition to the telemetry sample rate. For example, if the telemetry sample rate is 20% (default), the head sample rate is 1%, and the tail sample
+    /// rate is 15% (default), the effective sample rate will be 20% x 1% x 15% = 0.03%.
     ///
-    /// - Returns: A `MethodCalledTrace` instance to be used to stop the method call and measure it's execution time. It can be `nil` if the method call is not sampled.
+    /// Unlike the telemetry sample rate and tail-based sampling in `stopMethodCalled()`, this sample rate is applied at the start of the method call timing.
+    /// This head-based sampling reduces the impact of processing high-frequency metrics in the SDK core, as most samples can be dropped
+    /// before being passed to the message bus.
+    ///
+    /// - Returns: A `MethodCalledTrace` instance for stopping the method call and measuring its execution time, or `nil` if the method call is not sampled.
     func startMethodCalled(
         operationName: String,
         callerClass: String,
-        samplingRate: Float = 100.0
+        sampleRate: Float
```

Comment:
What is the reason behind removing the default value here?

Reply:
At first, I wanted compiler to guide me on all places where this is used 🙂, but after I was done I thought it's simply safer to not provide default for this particular call. The `startMethodCall()` is a high-frequency metric, defaulting it to 100% and overlooking it by mistake may bring large perf regression.